src/figuresClasses.ts

```diff
@@ -1,19 +1,78 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape
+  color: Color
+  getArea() : number
 }
 
 export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0
+      || b <= 0
+      || c <= 0
+      || Math.max(a, b, c) >= a + b + c - Math.max(a, b, c)) {
+      throw new Error('That isnt realy triangle');
+    }
+  }
+
+  getArea():number {
+    const s = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
 
+    return Math.floor(area * 100) / 100;
+  }
 }
 
 export class Circle implements Figure {
+  shape: Shape = 'circle';
 
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('That isnt realy circle');
+    }
+  }
+
+  getArea():number {
+    const area = Math.PI * this.radius * this.radius;
```

Comment:
This could be done with `Math.pow` or `**`
```suggestion
    const area = Math.PI * this.radius ** 2;
```